### PR TITLE
Chk 1808 mock npg fetch

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Add preauthorization api mock from payment-methods

The client-generated solution is not suitable since codegen didn't generate the client in the right way (see #53 )

#### List of Changes
Add mock business logic for new payment methods api
Invoke npg api using fetch
Get json response and build CreateSessionResponse from json data to mock response

#### Motivation and Context
Update mock for new npg integration

#### How Has This Been Tested?
Tests were performed locally using postman

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/handlers/ecommerce/payment-method.ts [hunk 1]
```diff
@@ -7,3 +8,63 @@ export const ecommerceGetPaymentMethods: RequestHandler = async (req, res) => {
...
+      "Correlation-Id": "df615d11-1159-43c9-bd07-fdc6f460e449",
+      "X-Api-key": X_API_KEY
+    },
+    hostname: "stg-ta.nexigroup.com",
+    maxRedirects: 20,
+    method: "POST",
+    path: "/api/phoenix-0.0/psp/api/v1/orders/build"
+  };
+
+  const build = https.request(options, response => {
```
Comment (giovanniberti) [inline src/handlers/ecommerce/payment-method.ts:30]: Why not use `fetch`?

File: src/handlers/ecommerce/payment-method.ts [hunk 2]
```diff
@@ -7,3 +8,63 @@ export const ecommerceGetPaymentMethods: RequestHandler = async (req, res) => {
...
+    response.on("end", () => {
+      const body = Buffer.concat(chunks);
+      res.status(200).send(body);
+    });
+
+    response.on("error", error => {
+      logger.error(error);
+    });
+  });
+  const postData = JSON.stringify({
```
Comment (giovanniberti) [inline src/handlers/ecommerce/payment-method.ts:48]: I would generate typings from the API spec as we do for other services

File: src/handlers/ecommerce/payment-method.ts (hunk 1 above)
Comment (ciuffagianluca) [reply src/handlers/ecommerce/payment-method.ts:30]: I started from #52 solution that implement this kind of request.

Comment (giovanniberti) [reply src/handlers/ecommerce/payment-method.ts:30]: I would still use `fetch` as it is less error prone and more readable

Comment (ciuffagianluca) [reply src/handlers/ecommerce/payment-method.ts]: did it!

File: src/app.ts [hunk 3]
```diff
@@ -205,6 +208,12 @@ export const newExpressApp: () => Promise<Express.Application> = async () => {
     ecommerceGetPspByPaymentMethods
   );
 
+  // payment-methods-service preauthorizations npg proxy
+  app.post(
+    "/ecommerce/checkout/v1/payment-methods/:id/preauthorizations",
```
Comment (infantesimone) [inline src/app.ts:213]: reminder: to update with sessions

Comment (ciuffagianluca) [reply src/app.ts:213]: Did it

File: src/handlers/ecommerce/payment-method.ts [hunk 4]
```diff
@@ -1,9 +1,82 @@
...
+import { CreateSessionResponse } from "../../generated/ecommerce_payment_methods/CreateSessionResponse";
+import { ProblemJson } from "../../generated/ecommerce_payment_methods/ProblemJson";
+import { Field } from "../../generated/ecommerce_payment_methods/Field";
 
 export const ecommerceGetPaymentMethods: RequestHandler = async (req, res) => {
   logger.info("[Get payment-methods ecommerce] - Return success case");
   res.status(200).send(createSuccessGetPaymentMethods());
 };
+
+const { X_API_KEY } = process.env;
```
Comment (Fernando-Granato) [inline src/handlers/ecommerce/payment-method.ts:17]: Add this variable X_API_KEY into env.example file. If possible let's give it a more meaningful name to define that it is NPG's api key 🙂

**Review by Fernando-Granato — APPROVED** [review]

**Review by infantesimone — APPROVED** [review]

**Pull request merged**